File: Java/src/main/java/QuestionBank/Solution001_100/Solution13.java
package QuestionBank.Solution001_100;

import java.util.HashMap;

/**
 * @author windows
 */
public class Solution13 {

    public int romanToInt(String s) {
        HashMap<Character, Integer> map = new HashMap<Character, Integer>(){{
            put('I', 1);
            put('V', 5);
            put('X', 10);
            put('L', 50);
            put('C', 100);
            put('D', 500);
            put('M', 1000);
        }};
        int sum = 0;
        for (int i = 0; i < s.length() - 1; i++) {
            sum += map.get(s.charAt(i));
            if (map.get(s.charAt(i)) < map.get(s.charAt(i + 1))){
                sum -= 2 * map.get(s.charAt(i));
            }
        }
        sum += map.get(s.charAt(s.length() - 1));
        return sum;
    }

    public static void main(String[] args) {
        Solution13 solution13 = new Solution13();
    }
}

File: C++/src/Solution1_100/Solution13.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution13.h"
#include <unordered_map>

int Solution13::romanToInt(string s) {
    int result = 0;
    unordered_map<char, int> romanMap = {{'I', 1}, {'V', 5}, {'X', 10}, {'L', 50}, {'C', 100}, {'D', 500}, {'M', 1000}};

    for (int i = 0; i < s.size(); ++i) {
        if (i < s.size() - 1 && romanMap[s[i]] < romanMap[s[i + 1]]) {
            result -= romanMap[s[i]];
        }else{
            result += romanMap[s[i]];
        }
    }
    return result;
}
